### PR TITLE
build: use ng-packagr to generate library

### PR DESCRIPTION
**Pull request by trotyl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Switch to [ng-packagr](https://github.com/dherges/ng-packagr) to build package which follows [Angular Package Format 6.0](https://docs.google.com/document/d/1CZC2rcpxffTDfRDs6p1cfbmKNLA6x5O-NtkJglDaBVs/preview).

Highlights:

+ ~~`dependencies` is not allowed by `ng-packagr`, only `peerDependencies` can be used;~~
+ `import * as pkg from 'pkg'` are now `import pkg from 'pkg'`;
+ Root `package.json` no longer used to publish, only for dev purpose now;
+ Package sub-path no longer contains the `src` segment;

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

CSS file imports require remove the `/src` segment.

### CODE REVIEW

File: components/tsconfig.json [hunk 1]
```diff
@@ -15,6 +17,12 @@
     "./ng-zorro-antd.module.ts"
   ],
   "angularCompilerOptions": {
-    "skipTemplateCodegen": true
+    "annotateForClosureCompiler": true,
+    "skipTemplateCodegen": true,
+    "strictMetadataEmit": true,
+    "fullTemplateTypeCheck": false,
```
Comment (trotyl) [inline components/tsconfig.json]: This should be enabled once our lib could pass it.

**Comment by trotyl** [comment]
Previously I thought the `import * as` -> `import default` change could break type-checking without `skipLibCheck` or `allowSyntheticDefaultImports`, but turns out it would still be fine.

**Comment by trotyl** [comment]
Found the `whitelistedNonPeerDependencies` option which allows `dependencies`.

**Comment by vthinkxie** [comment]
Great, it seems there are no break changes now

**Comment by trotyl** [comment]
@vthinkxie The CSS path is still breaking, previously:

```
node_modules/ng-zorro-antd/src/ng-zorro-antd.min.css
```

Now:

```
node_modules/ng-zorro-antd/ng-zorro-antd.min.css
```

**Pull request merged**